Core/src/IfcGeometryConverter/CurveConverter.h

```diff
@@ -2107,106 +2010,151 @@ namespace OpenInfraPlatform {
-				/*! \brief Calculates a trimming point on the curve using \c IfcCartesianPoint. .
-				* \param[in] circle					A pointer to data from \c IfcCircle.
+				/*! \brief Calculates a trimming point on the circle using \c IfcCartesianPoint.
+				* \param[in] circle					A pointer to data from a \c IfcCircle.
 				* \param[in] cartesianPoint			A pointer to data form \c IfcCartesianPoint.
 				* \return							The location of the trimming point.
 				*/
 				carve::geom::vector<3> getPointOnCurve(const EXPRESSReference<typename IfcEntityTypesT::IfcCircle>& circle,
 					const EXPRESSReference<typename IfcEntityTypesT::IfcCartesianPoint>& cartesianPoint) const throw(...)
 				{
+					carve::math::Matrix conicPositionMatrix = placementConverter->convertIfcAxis2Placement(circle->Position);
+					carve::geom::vector<3> conicCenter = conicPositionMatrix * carve::geom::VECTOR(0., 0., 0.);
 
-					return placementConverter->convertIfcCartesianPoint(cartesianPoint);
+					return conicCenter + placementConverter->convertIfcCartesianPoint(cartesianPoint);
 				}
 
 				/**********************************************************************************************/
-				/*! \brief Calculates a trimming point on the curve using \c IfcParameterValue.
-				* \param[in] circle					A pointer to data from \c IfcCircle.
-				* \param[in] parameter				A pointer to data form \c IfcParameterValue.
+				/*! \brief Calculates a trimming point on the ellipse using \c IfcCartesianPoint.
+				* \param[in] ellipse				A pointer to data from a \c IfcEllipse.
+				* \param[in] cartesianPoint			A pointer to data form \c IfcCartesianPoint.
 				* \return							The location of the trimming point.
 				*/
-				carve::geom::vector<3> getPointOnCurve(const EXPRESSReference<typename IfcEntityTypesT::IfcCircle>& circle,
-					const typename IfcEntityTypesT::IfcParameterValue & parameter) const throw(...)
+				carve::geom::vector<3> getPointOnCurve(const EXPRESSReference<typename IfcEntityTypesT::IfcEllipse>& ellipse,
+					const EXPRESSReference<typename IfcEntityTypesT::IfcCartesianPoint>& cartesianPoint) const throw(...)
 				{
-					double angle = parameter * UnitConvert()->getAngleInRadianFactor();
-					// determine position
-					carve::math::Matrix conicPositionMatrix = placementConverter->convertIfcAxis2Placement(circle->Position);
-
-					// Get radius
-					double circleRadius = circle->Radius * UnitConvert()->getLengthInMeterFactor();
+					carve::math::Matrix conicPositionMatrix = placementConverter->convertIfcAxis2Placement(ellipse->Position);
+					carve::geom::vector<3> conicCenter = conicPositionMatrix * carve::geom::VECTOR(0., 0., 0.);
 
-					carve::geom::vector<3> circleCenter =
-						conicPositionMatrix * carve::geom::VECTOR(0, 0, 0);
-					return circleCenter + carve::geom::VECTOR(circleRadius * cos(angle), circleRadius * sin(angle), 0.);
+					return conicCenter + placementConverter->convertIfcCartesianPoint(cartesianPoint);

```

Comment:
I hope, I have implemented it right in 7a3792f. I have calculated this as (left-)multiplication (At least, I think so). Right-multiplication was not possible, since the `return` should be then from type `carve::math::Matrix`

Reply:
Resolved: 7826079

Reply:
Resolved: 7826079